src/binding-factory.ts

```diff
@@ -23,9 +23,17 @@ const luaBindings: Record<string, luaBindingFactoryFunc> = {
     },
     "5.0.x": function(luaGlue: LuaEmscriptenModule){
         return {
+            // #define lua_getglobal(L,s)  lua_getfield(L, LUA_GLOBALSINDEX, s)
+            lua_getglobal: function (L: LuaState, name: string) {
+                return (this as Lua).lua_getfield(L, LUA_GLOBALSINDEX_50, name);
+            },
             lua_getfield: function(L: LuaState, index: number, k: string) {
                 (this as Lua).lua_pushstring(L, k);
-                return (this as Lua).lua_gettable(L, index);
+
+                // If the index is a relative offset, it needs to be corrected for the grown stack
+                const isOffset = index < 0 && index !== LUA_GLOBALSINDEX_50;
```

Comment:
Good catch! Instead of the comment i would prefer an additional variable. Something like this:
```suggestion
                const isRelativeOffset = index < 0;
                const shouldCorrectIndex= isRelativeOffset && index !== LUA_GLOBALSINDEX_50;
```

Reply:
How about just renaming `isOffset` to `isRelativeOffset`? The globals index is a pseudo-index, which is by definition not a relative offset, even if it is negative. 😁

Reply:
Sounds good!

Reply:
Great, how does this look?